Sources/Subprocess/Platforms/Subprocess+Unix.swift

```
@@ -397,67 +392,32 @@ extension FileDescriptor {
 internal typealias PlatformFileDescriptor = CInt
 internal typealias TrackedPlatformDiskIO = TrackedDispatchIO
 
-extension CreatedPipe {
-    internal func createInputPipe() -> InputPipe {
-        var writeEnd: TrackedPlatformDiskIO? = nil
-        if let writeFileDescriptor = self.writeFileDescriptor {
-            let dispatchIO: DispatchIO = DispatchIO(
-                type: .stream,
-                fileDescriptor: writeFileDescriptor.platformDescriptor,
-                queue: .global(),
-                cleanupHandler: { error in
-                    // Close the file descriptor
-                    if writeFileDescriptor.closeWhenDone {
-                        try? writeFileDescriptor.safelyClose()
-                    }
+extension TrackedFileDescriptor {
+    internal consuming func createPlatformDiskIO() -> TrackedPlatformDiskIO {
+        let dispatchIO: DispatchIO = DispatchIO(
+            type: .stream,
+            fileDescriptor: self.platformDescriptor(),
+            queue: .global(),
+            cleanupHandler: { error in
+                // Close the file descriptor
+                if self.closeWhenDone {
+                    try? self.safelyClose()
```

why ignore the error? That can't be quite right